Add rescan for datastores, convert to python2

ekhomyakova opened the pull request.

Commit:
Add rescan for datastores, convert to python2

ekhomyakova commented:
```
$ python victl.py -h
usage: victl.py [-h]
                {check-dvs-attached,datastore-list,rescan-datastores,check-portgroup,cluster-list,check-esxi,check-datastore}
                ...

positional arguments:
  {check-dvs-attached,datastore-list,rescan-datastores,check-portgroup,cluster-list,check-esxi,check-datastore}

optional arguments:
  -h, --help            show this help message and exit

    -----------------------------------
    You can use these environment variables:
        VCENTER_IP         --host
        VCENTER_PASSWORD   --password
        VCENTER_USERNAME   --user
        VC_CLUSTER         --cluster
        VC_DATACENTER      --datacenter
        VC_DATASTORE       --datastore

    -----------------------------------
    Examples of usage:
        victl.py check-datastore -c 'Cluster1' -d 'Datacenter' -ds 'nfs' -s '172.16.0.254' -p 'Qwer!1234' -o '443' -u 'administrator@vsphere.local'

        victl.py check-dvs-attached -c 'Cluster1' -d 'Datacenter' -s '172.16.0.254' -p 'Qwer!1234' -o '443' -u 'administrator@vsphere.local' -v 'dvSwitch' -n 'vmnic1'

        victl.py check-esxi -c 'Cluster1' -d 'Datacenter' -s '172.16.0.254' -p 'Qwer!1234' -o '443' -sp 'swordfish' -su 'root' -u 'administrator@vsphere.local'

        victl.py check-portgroup -c 'Cluster1' -d 'Datacenter' -s '172.16.0.254' -p 'Qwer!1234' -o '443' -g 'br100' -u 'administrator@vsphere.local'

        victl.py cluster-list -d 'Datacenter' -s '172.16.0.254' -p 'Qwer!1234' -o '443' -u 'administrator@vsphere.local'

        victl.py datastore-list -c 'Cluster1' -d 'Datacenter' -s '172.16.0.254' -p 'Qwer!1234' -o '443' -u 'administrator@vsphere.local'

        victl.py rescan-datastores -d 'Datacenter' -s '172.16.0.254' -p 'Qwer!1234' -o '443' -u 'administrator@vsphere.local'

```

naumvd95 reviewed (approved):
need to test on CI

naumvd95 committed:
Merge branch 'master' into rescan

naumvd95 committed:
testing vcenter-wrapper case

+this is quick-fix (need to review and discuss possibilities in writing
normal,beauty fix)
+by the way, specific vcenter parameters redefined in 9.0 vcenter and
9.x vcenter jobs
+now need to start nightly on 784 snapshot

rkhozinov commented on servers/partner-ci/builders/workstation.sh:
```diff
@@ -139,6 +139,15 @@ configure_nfs(){
+    BACKUP_IFS=$IFS
+    IFS=','
+    for cluster in $VCENTER_CLUSTERS; do
+      python /btsync/victl.py rescan-datastores -d 'Datacenter' -s '172.16.0.145' -p 'vmware' -o '443' -u 'root'
```
Should be used as env vars.
VCENTER_IP
VCENTER_USERNAME
VCENTER_PASSWORD

rkhozinov replied:
victl should be integrated to job, don't use external dependencies like /btsync and so on

rkhozinov replied:
ok

rkhozinov commented on servers/partner-ci/builders/workstation.sh:
```diff
@@ -139,6 +139,15 @@ configure_nfs(){
     esxi_exec $esxi_host 'esxcli storage core adapter rescan --all'
```
we don't need that

rkhozinov commented on servers/partner-ci/builders/workstation.sh:
```diff
@@ -139,6 +139,15 @@ configure_nfs(){
     esxi_exec $esxi_host 'esxcli storage core adapter rescan --all'
     esxi_exec $esxi_host 'esxcli storage nfs list'
```
and that
if we'll use victl

rkhozinov reviewed (changes requested):
Separate vcenter fix and nfs rescan with different pull requests

naumvd95 committed:
defining values as variables

rkhozinov committed:
Merge branch 'master' into rescan

naumvd95 commented:
@rkhozinov  recheck please

rkhozinov commented on servers/partner-ci/9.x/vcenter.yaml:
```diff
@@ -10,6 +10,7 @@
             CONV_SNAP_PATH=utils/jenkins/conv_snapshot_file.py;
             DISABLE_STP=true;
             ESXI_HOSTS=172.16.0.250 172.16.0.252 172.16.0.253;
+            FORCE_DISABLE_UPDATES=true;
```
Why?

rkhozinov commented on servers/partner-ci/builders/common.prepare.sh:
```diff
@@ -91,6 +91,9 @@ function get_venv_requirements {
   export SPEC_REQS_PATH="${WORKSPACE}/plugin_test/requirement.txt"
   sed -i 's/2.9.23/release\/2.9/g' $REQS_PATH_DEVOPS
+  # additional libraries associated with vcenter-control wrapper
+  echo "pyvim" >> $REQS_PATH
```
echo -e "pyvim\n pyvmomi"

rkhozinov commented on servers/partner-ci/builders/workstation.sh:
```diff
@@ -136,8 +136,15 @@ configure_nfs(){
+    BACKUP_IFS=$IFS
+    IFS=','
+    for cluster in $VCENTER_CLUSTERS; do
+      python /btsync/victl.py rescan-datastores -d 'Datacenter' -s $VCENTER_IP -p $VCENTER_PASSWORD -o '443' -u $VCENTER_USERNAME
```
Find a way to avoid additional dependencies like /btsync and so on

rkhozinov committed:
Merge branch 'master' into rescan

Commit:
Add script for datastores rescan

Commit:
Merge pull request #148 from rkhozinov/victl

Add script for datastores rescan

naumvd95 committed:
fast-exec file for rescan datastores attached

testing on dvs
result success
http://jenkins-tpi.bud.mirantis.net:8080/view/9.x.dvs/job/9.x.dvs.custom.qa/76/console
@rkhozinov need to approve
and sync other jobs to associate with 9.x.dvs

rkhozinov commented on servers/partner-ci/builders/workstation.sh:
```diff
@@ -137,14 +137,14 @@ configure_nfs(){
-      python /btsync/victl.py datastore-list -c $cluster -d 'Datacenter' -s $VCENTER_IP -p $VCENTER_PASSWORD -o '443' -u $VCENTER_USERNAME
-    done
-    IFS=$BACKUP_IFS
+   # #vcenter-control wrapper
```
remove commented lines

naumvd95 replied:
done

naumvd95 committed:
remove useless code

rkhozinov committed:
Merge branch 'master' into rescan

naumvd95 committed:
merge all jobs to sync with new rescan-ds-case

rkhozinov commented on servers/partner-ci/builders/rescan_ds.py:
```diff
@@ -0,0 +1,410 @@
+                return dc
+
+        raise NotFoundException("Can not find dc "
+                                "'{dc_name}'".format(dc_name=datacenter))
```
we have a specific check for jjb configuration to avoid unresolved jjb's variables, like {var1}. So we need to change template for python's .format() to {} and {0}, {1}, etc.

naumvd95 committed:
fix code-style formats to pass travis checks

naumvd95 committed:
test regex mk1

naumvd95 committed:
testing regex mk2

naumvd95 committed:
travis: grep checker is updated

Commit:
rescan_datastores: refactoring

Commit:
rescan: refactoring

Commit:
rescan_ds: add a fix to run with virtualenv libs

rkhozinov committed:
Update rescan_ds.py

Commit:
Merge branch 'master' into rescan-refactoring

Conflicts:
	.travis.yml

Commit:
rescan_ds: style fixes related to flake8

naumvd95 committed:
fix to avoid venv-activating during python-build-step

proofs
http://jenkins-tpi.bud.mirantis.net:8080/view/9.x.dvs/job/9.x.dvs.custom.qa/100/console

naumvd95 committed:
Merge branch 'rescan-refactoring' of github.com:rkhozinov/jenkins-jobs into rescan-refactoring

rkhozinov committed:
travis: fix checking of undefined jjb variables

naumvd95 committed:
refactoring venv-system, restructure rescan nfs case

1)direct injecting venv_path as a common var for all jobs
2)activating venv from python step
3) bug-fixes
proofs
http://jenkins-tpi.bud.mirantis.net:8080/view/9x-custom/job/9.x.vcenter.custom.qa/94/console

naumvd95 committed:
Merge branch 'rescan-refactoring' of github.com:rkhozinov/jenkins-jobs into rescan-refactoring

naumvd95 committed:
refactor-cases , as @rkhozinov decision

naumvd95 committed:
code-style fix to avoid travis checks

Commit:
Refactoring according with pep8

Commit:
Fix venv activating

rkhozinov committed:
Merge pull request #150 from rkhozinov/rescan-refactoring

rescan_datastores: refactoring

naumvd95 committed:
Merge branch 'master' into rescan
